The log file for 5 days has reached an astonishing 8G.

### Summary

```
drwxr-xr-x david david 236 B  Wed Feb 26 16:41:44 2025  .
drwxr-xr-x david david 158 B  Wed Feb 26 11:37:58 2025  ..
.rw-r--r-- david david 2.0 KB Wed Feb 26 11:37:58 2025  telemetry.log
.rw-r--r-- david david 557 B  Tue Feb 18 16:21:23 2025  'zed-2025_02_18 08_21_23.panic'
.rw-r--r-- david david 603 B  Fri Feb 21 16:35:39 2025  'zed-2025_02_21 08_35_39.panic'
.rw-r--r-- david david 556 B  Sun Feb 23 18:20:23 2025  'zed-2025_02_23 10_20_23.panic'
.rw-r--r-- david david 120 KB Wed Feb 26 11:38:13 2025  Zed.log
.rw-r--r-- david david 7.9 GB Tue Feb 25 19:10:23 2025 󰁯 Zed.log.old
```

Not sure exactly what happened. But looking at the log content, in about five days, the log file reached about 8GB. File content:

```
2025-02-21T22:49:20.96907153+08:00 [INFO] Using surface present mode MAILBOX
2025-02-21T22:49:20.969131205+08:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-02-21T22:49:21.013757028+08:00 [INFO] Using surface present mode MAILBOX
2025-02-21T22:49:21.013826215+08:00 [INFO] Retaining current format: Bgra8UnormSrgb
2025-02-21T22:50:18.863739425+08:00 [ERROR] window not found
2025-02-21T22:50:18.863802882+08:00 [ERROR] window not found
2025-02-21T22:50:54.298086791+08:00 [INFO] ========== starting zed ==========
2025-02-21T22:50:54.568101557+08:00 [WARN] failed to load font 'SymbolsNFM'
2025-02-21T22:50:54.593381983+08:00 [WARN] failed to load font 'SymbolsNFM'
2025-02-21T22:50:54.595077313+08:00 [INFO] Compositor detection: _NET_WM_CM_S?=false, _NET_WM_CM_OWNER=false, _NET_SUPPORTING_WM_CHECK=true
2025-02-21T22:50:54.595137542+08:00 [INFO] x11: compositor present: true, gtk_frame_extents_supported: true
2025-02-21T22:50:54.625329506+08:00 [INFO] Presentation support: "VK_KHR_wayland_surface"
2025-02-21T22:50:54.625388581+08:00 [INFO] Presentation support: "VK_KHR_xcb_surface"
2025-02-21T22:50:54.625415664+08:00 [INFO] Presentation support: "VK_KHR_xlib_surface"
2025-02-21T22:50:54.625444008+08:00 [INFO] Enabling Vulkan Portability
2025-02-21T22:50:54.625473117+08:00 [INFO] Enabling color space support
2025-02-21T22:50:54.630256715+08:00 [INFO] Adapter: "Intel(R) Arc(tm) Graphics (MTL)"
2025-02-21T22:50:54.630353469+08:00 [INFO] Ray tracing is supported
2025-02-21T22:50:54.633893898+08:00 [INFO] Try connect ibus
2025-02-21T22:50:54.634889905+08:00 [INFO] perform;
2025-02-21T22:50:54.634956345+08:00 [INFO] authenticate;
2025-02-21T22:50:54.635033237+08:00 [INFO] read_command;
2025-02-21T22:50:54.635114615+08:00 [INFO] send_secondary_commands;
2025-02-21T22:50:54.635210723+08:00 [INFO] Opening main db
2025-02-21T22:50:54.635630037+08:00 [INFO] new;
2025-02-21T22:50:54.635686744+08:00 [INFO] socket reader;
2025-02-21T22:50:54.636201309+08:00 [INFO] keep_updated;
2025-02-21T22:50:54.636585875+08:00 [INFO] Opening main db
...
...
...
2025-02-25T19:10:23.762163018+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762201375+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762236852+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762271647+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762312304+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762361926+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762407263+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762443965+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762478146+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762512267+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762546584+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762581215+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762615364+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762650538+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762684551+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.76271916+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.76275342+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762808735+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762849873+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762887869+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762923352+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762956715+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.762990426+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763027986+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763062441+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.76309708+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763131121+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763171346+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763213151+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763247566+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763281845+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763315951+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763359422+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763393594+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763427685+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763462005+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763496265+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763530332+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.76356497+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763598115+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763631495+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763665186+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763698578+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763732191+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763765636+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763799104+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763832577+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763867902+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763901915+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.763936129+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.7639706+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764005021+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764039328+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764073767+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764108508+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764144044+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764181258+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764214861+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764256455+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764303576+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764351908+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764400239+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764477684+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764556839+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764640326+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764721636+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764799171+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764860675+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764934089+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.764986098+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765023256+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765058758+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765098162+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765141608+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765179254+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765212366+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765249952+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765306528+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765361477+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765396164+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765549546+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765655602+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765754258+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765852084+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.765950408+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.766047708+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.766145548+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.766243514+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.766377741+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.766477083+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.76657593+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.766675766+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.766774572+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.766873986+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.767004275+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.767104137+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.76720316+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.767299942+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
2025-02-25T19:10:23.767404661+08:00 [WARN] error polling for X11 events: IoError(Custom { kind: Other, error: UnknownError })
```

I didn't open the file. However, I investigated the file content through cat. It seems that it was mainly generated on the 25th.



![Image](https://github.com/user-attachments/assets/5efa7f3f-6ddd-4fa6-bacb-b85d64c4e47b)

Is this normal?

### Zed Version and System Specs

Zed: v0.174.8 (Zed) OS: Linux X11 arch unknown Memory: 30.8 GiB Architecture: x86_64 GPU: Intel(R) Arc(tm) Graphics (MTL) || Intel open-source Mesa driver || Mesa 24.3.4-arch1.1

--- a/crates/zed/src/logger.rs
+++ b/crates/zed/src/logger.rs
@@ -1,0 +1,124 @@
+use chrono::Offset;
+use env_logger::Builder;
+use log::LevelFilter;
+use simplelog::ConfigBuilder;
+use std::fs::{self, File, OpenOptions};
+use std::io::{self, Write};
+use std::path::PathBuf;
+use time::UtcOffset;
+
+pub fn init_logger() {
+    let level = LevelFilter::Info;
+
+    // Prevent log file from becoming too large.
+    const KIB: u64 = 1024;
+    const MIB: u64 = 1024 * KIB;
+    const MAX_LOG_BYTES: u64 = MIB;
+    if std::fs::metadata(paths::log_file()).map_or(false, |metadata| metadata.len() > MAX_LOG_BYTES)
+    {
+        let _ = std::fs::rename(paths::log_file(), paths::old_log_file());
+    }
+
+    match CustomWriter::new(paths::log_file(), paths::old_log_file(), MAX_LOG_BYTES) {
+        Ok(writer) => {
+            let mut config_builder = ConfigBuilder::new();
+
+            config_builder.set_time_format_rfc3339();
+            let local_offset = chrono::Local::now().offset().fix().local_minus_utc();
+            if let Ok(offset) = UtcOffset::from_whole_seconds(local_offset) {
+                config_builder.set_time_offset(offset);
+            }
+
+            #[cfg(any(target_os = "linux", target_os = "freebsd"))]
+            {
+                config_builder.add_filter_ignore_str("zbus");
+                config_builder.add_filter_ignore_str("blade_graphics::hal::resource");
+                config_builder.add_filter_ignore_str("naga::back::spv::writer");
+            }
+
+            let config = config_builder.build();
+            simplelog::WriteLogger::init(level, config, writer)
+                .expect("could not initialize logger");
+        }
+        Err(err) => {
+            init_stdout_logger();
+            log::error!(
+                "could not open log file, defaulting to stdout logging: {}",
+                err
+            );
+        }
+    }
+}
+
+pub fn init_stdout_logger() {
+    Builder::new()
+        .parse_default_env()
+        .format(|buf, record| {
+            use env_logger::fmt::style::{AnsiColor, Style};
+
+            let subtle = Style::new().fg_color(Some(AnsiColor::BrightBlack.into()));
+            write!(buf, "{subtle}[{subtle:#}")?;
+            write!(
+                buf,
+                "{} ",
+                chrono::Local::now().format("%Y-%m-%dT%H:%M:%S%:z")
+            )?;
+            let level_style = buf.default_level_style(record.level());
+            write!(buf, "{level_style}{:<5}{level_style:#}", record.level())?;
+            if let Some(path) = record.module_path() {
+                write!(buf, " {path}")?;
+            }
+            write!(buf, "{subtle}]{subtle:#}")?;
+            writeln!(buf, " {}", record.args())
+        })
+        .init();
+}
+
+struct CustomWriter {
+    file: File,
+    path: &'static PathBuf,
+    old_path: &'static PathBuf,
+    max_size: u64,
+    current_size: u64,
+}
+
+impl CustomWriter {
+    fn new(path: &'static PathBuf, old_path: &'static PathBuf, max_size: u64) -> io::Result<Self> {
+        let file = OpenOptions::new().create(true).append(true).open(&path)?;
+        let current_size = file.metadata()?.len();
+
+        Ok(CustomWriter {
+            file,
+            path,
+            old_path,
+            max_size,
+            current_size,
+        })
+    }
+
+    fn swap(&mut self) -> io::Result<()> {
+        self.file.sync_all()?;
+        fs::rename(&self.path, &self.old_path)?;
+        self.file = OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(&self.path)?;
+        self.current_size = 0;
+        Ok(())
+    }
+}
+
+impl Write for CustomWriter {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        if self.current_size + buf.len() as u64 > self.max_size {
+            self.swap()?;
+        }
+        let bytes = self.file.write(buf)?;
+        self.current_size += bytes as u64;
+        Ok(bytes)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        self.file.flush()
+    }
+}

--- a/crates/zed/src/logger.rs
+++ b/crates/zed/src/logger.rs
@@ -4,7 +4,6 @@ use log::LevelFilter;
 use simplelog::ConfigBuilder;
 use std::fs::{self, File, OpenOptions};
 use std::io::{self, Write};
-use std::path::PathBuf;
 use time::UtcOffset;
 
 pub fn init_logger() {
@@ -19,7 +18,7 @@ pub fn init_logger() {
         let _ = std::fs::rename(paths::log_file(), paths::old_log_file());
     }
 
-    match CustomWriter::new(paths::log_file(), paths::old_log_file(), MAX_LOG_BYTES) {
+    match LogWriter::new(MAX_LOG_BYTES) {
         Ok(writer) => {
             let mut config_builder = ConfigBuilder::new();
 
@@ -74,44 +73,43 @@ pub fn init_stdout_logger() {
         .init();
 }
 
-struct CustomWriter {
+struct LogWriter {
     file: File,
-    path: &'static PathBuf,
-    old_path: &'static PathBuf,
     max_size: u64,
     current_size: u64,
 }
 
-impl CustomWriter {
-    fn new(path: &'static PathBuf, old_path: &'static PathBuf, max_size: u64) -> io::Result<Self> {
-        let file = OpenOptions::new().create(true).append(true).open(&path)?;
+impl LogWriter {
+    fn new(max_size: u64) -> io::Result<Self> {
+        let file = OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(paths::log_file())?;
         let current_size = file.metadata()?.len();
 
-        Ok(CustomWriter {
+        Ok(LogWriter {
             file,
-            path,
-            old_path,
             max_size,
             current_size,
         })
     }
 
-    fn swap(&mut self) -> io::Result<()> {
+    fn replace(&mut self) -> io::Result<()> {
         self.file.sync_all()?;
-        fs::rename(&self.path, &self.old_path)?;
+        fs::rename(paths::log_file(), paths::old_log_file())?;
         self.file = OpenOptions::new()
             .create(true)
             .append(true)
-            .open(&self.path)?;
+            .open(paths::log_file())?;
         self.current_size = 0;
         Ok(())
     }
 }
 
-impl Write for CustomWriter {
+impl Write for LogWriter {
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
         if self.current_size + buf.len() as u64 > self.max_size {
-            self.swap()?;
+            self.replace()?;
         }
         let bytes = self.file.write(buf)?;
         self.current_size += bytes as u64;

--- a/crates/zed/src/main.rs
+++ b/crates/zed/src/main.rs
@@ -1,11 +1,11 @@
 // Disable command line from opening on release mode
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
 
+mod logger;
 mod reliability;
 mod zed;
 
 use anyhow::{anyhow, Context as _, Result};
-use chrono::Offset;
 use clap::{command, Parser};
 use cli::FORCE_CLI_MODE_ENV_VAR_NAME;
 use client::{parse_zed_link, Client, ProxySettings, UserStore};
@@ -13,7 +13,6 @@ use collab_ui::channel_view::ChannelView;
 use collections::HashMap;
 use db::kvp::{GLOBAL_KEY_VALUE_STORE, KEY_VALUE_STORE};
 use editor::Editor;
-use env_logger::Builder;
 use extension::ExtensionHostProxy;
 use extension_host::ExtensionStore;
 use fs::{Fs, RealFs};
@@ -24,11 +23,11 @@ use gpui::{App, AppContext as _, Application, AsyncApp, UpdateGlobal as _};
 use gpui_tokio::Tokio;
 use http_client::{read_proxy_from_env, Uri};
 use language::LanguageRegistry;
-use log::LevelFilter;
 use prompt_library::PromptBuilder;
 use reqwest_client::ReqwestClient;
 
 use assets::Assets;
+use logger::{init_logger, init_stdout_logger};
 use node_runtime::{NodeBinaryOptions, NodeRuntime};
 use parking_lot::Mutex;
 use project::project_settings::ProjectSettings;
@@ -36,11 +35,9 @@ use recent_projects::{open_ssh_project, SshSettings};
 use release_channel::{AppCommitSha, AppVersion, ReleaseChannel};
 use session::{AppSession, Session};
 use settings::{watch_config_file, Settings, SettingsStore};
-use simplelog::ConfigBuilder;
 use std::{
     env,
-    fs::OpenOptions,
-    io::{self, IsTerminal, Write},
+    io::{self, IsTerminal},
     path::{Path, PathBuf},
     process,
     sync::Arc,
@@ -49,7 +46,6 @@ use theme::{
     ActiveTheme, IconThemeNotFoundError, SystemAppearance, ThemeNotFoundError, ThemeRegistry,
     ThemeSettings,
 };
-use time::UtcOffset;
 use util::{maybe, ResultExt, TryFutureExt};
 use uuid::Uuid;
 use welcome::{show_welcome_view, BaseKeymap, FIRST_OPEN};
@@ -198,7 +194,11 @@ fn main() {
         return;
     }
 
-    init_logger();
+    if stdout_is_a_pty() {
+        init_stdout_logger();
+    } else {
+        init_logger();
+    }
 
     log::info!("========== starting zed ==========");
 
@@ -934,82 +934,6 @@ fn init_paths() -> HashMap<io::ErrorKind, Vec<&'static Path>> {
         }
         errors
     })
-}
-
-fn init_logger() {
-    if stdout_is_a_pty() {
-        init_stdout_logger();
-    } else {
-        let level = LevelFilter::Info;
-
-        // Prevent log file from becoming too large.
-        const KIB: u64 = 1024;
-        const MIB: u64 = 1024 * KIB;
-        const MAX_LOG_BYTES: u64 = MIB;
-        if std::fs::metadata(paths::log_file())
-            .map_or(false, |metadata| metadata.len() > MAX_LOG_BYTES)
-        {
-            let _ = std::fs::rename(paths::log_file(), paths::old_log_file());
-        }
-
-        match OpenOptions::new()
-            .create(true)
-            .append(true)
-            .open(paths::log_file())
-        {
-            Ok(log_file) => {
-                let mut config_builder = ConfigBuilder::new();
-
-                config_builder.set_time_format_rfc3339();
-                let local_offset = chrono::Local::now().offset().fix().local_minus_utc();
-                if let Ok(offset) = UtcOffset::from_whole_seconds(local_offset) {
-                    config_builder.set_time_offset(offset);
-                }
-
-                #[cfg(any(target_os = "linux", target_os = "freebsd"))]
-                {
-                    config_builder.add_filter_ignore_str("zbus");
-                    config_builder.add_filter_ignore_str("blade_graphics::hal::resource");
-                    config_builder.add_filter_ignore_str("naga::back::spv::writer");
-                }
-
-                let config = config_builder.build();
-                simplelog::WriteLogger::init(level, config, log_file)
-                    .expect("could not initialize logger");
-            }
-            Err(err) => {
-                init_stdout_logger();
-                log::error!(
-                    "could not open log file, defaulting to stdout logging: {}",
-                    err
-                );
-            }
-        }
-    }
-}
-
-fn init_stdout_logger() {
-    Builder::new()
-        .parse_default_env()
-        .format(|buf, record| {
-            use env_logger::fmt::style::{AnsiColor, Style};
-
-            let subtle = Style::new().fg_color(Some(AnsiColor::BrightBlack.into()));
-            write!(buf, "{subtle}[{subtle:#}")?;
-            write!(
-                buf,
-                "{} ",
-                chrono::Local::now().format("%Y-%m-%dT%H:%M:%S%:z")
-            )?;
-            let level_style = buf.default_level_style(record.level());
-            write!(buf, "{level_style}{:<5}{level_style:#}", record.level())?;
-            if let Some(path) = record.module_path() {
-                write!(buf, " {path}")?;
-            }
-            write!(buf, "{subtle}]{subtle:#}")?;
-            writeln!(buf, " {}", record.args())
-        })
-        .init();
 }
 
 fn stdout_is_a_pty() -> bool {

